hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -644,14 +644,17 @@ public final Stream<FileSlice> getLatestUnCompactedFileSlices(String partitionSt
 
   @Override
   public final Stream<FileSlice> getLatestFileSlicesBeforeOrOn(String partitionStr, String maxCommitTime,
-      boolean includeFileSlicesInPendingCompaction) {
+      boolean includeFileSlicesInPendingCompaction, boolean includeFilesInPendingCompaction) {
     try {
       readLock.lock();
       String partitionPath = formatPartitionKey(partitionStr);
       ensurePartitionLoadedCorrectly(partitionPath);
       Stream<FileSlice> fileSliceStream = fetchLatestFileSlicesBeforeOrOn(partitionPath, maxCommitTime)
           .filter(slice -> !isFileGroupReplacedBeforeOrOn(slice.getFileGroupId(), maxCommitTime));
       if (includeFileSlicesInPendingCompaction) {
+        if (includeFilesInPendingCompaction) {
+          return fileSliceStream.map(this::addBootstrapBaseFileIfPresent);
```

Comment:
Can you explain a little more the details ? BTW, we should not modify the interfaces of fs view in most of the cases.

Reply:
thx for your review, @danny0405 
from the TM log:
2022-04-25 14:21:25,198 INFO  org.apache.hudi.common.table.view.AbstractTableFileSystemView - Pending Compaction instant for (FileSlice {fileGroupId=HoodieFileGroupId{partitionPath='', fileId='02d71f2b-bb2f-4c44-96f2-78cb791da28e'}, baseCommitTime=20220425131449, baseFile='null', logFiles='[HoodieLogFile{pathStr='hdfs://hacluster/tmp/hudi/hudi_sink/.02d71f2b-bb2f-4c44-96f2-78cb791da28e_20220425131449.log.1_4-50-0', fileLen=20884490}]'}) **is :Option**{val=(20220425131449,CompactionOperation{baseInstantTime='20220425131251', dataFileCommitTime=Option{val=20220425131251}, deltaFileNames=[.02d71f2b-bb2f-4c44-96f2-78cb791da28e_20220425131251.log.1_4-50-0], dataFileName=Option{val=02d71f2b-bb2f-4c44-96f2-78cb791da28e_8-10-0_20220425131251.parquet}, id='HoodieFileGroupId{partitionPath='', fileId='02d71f2b-bb2f-4c44-96f2-78cb791da28e'}', metrics={TOTAL_LOG_FILES=1.0, TOTAL_IO_READ_MB=84.0, TOTAL_LOG_FILES_SIZE=2.2786803E7, TOTAL_IO_WRITE_MB=62.0, TOTAL_IO_MB=146.0}, bootstrapFilePath=Optional.empty})}
2022-04-25 14:21:25,198 INFO  org.apache.hudi.common.table.view.AbstractTableFileSystemView - File Slice (FileSlice {fileGroupId=HoodieFileGroupId{partitionPath='', fileId='02d71f2b-bb2f-4c44-96f2-78cb791da28e'}, baseCommitTime=20220425131449, baseFile='null', logFiles='[HoodieLogFile{pathStr='hdfs://hacluster/tmp/hudi/hudi_sink/.02d71f2b-bb2f-4c44-96f2-78cb791da28e_20220425131449.log.1_4-50-0', fileLen=20884490}]'}) is in pending compaction

Reply:
You mean the `includeFileSlicesInPendingCompaction` does not work for pending compaction files ?

Reply:
When job is restarted, compaction operator and bootstrap operator are initialized at the same time. The bootstrapOperator#initializeState is executed before the compactionOperator#open. So bootstrapOperator#loadRecords may have got the wrong fileSlices, the logic (https://github.com/apache/hudi/blob/4258a715174e0a97271112148ab20ef9307e2bd8/hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java#L416) filters out pendingCompaction files

Reply:
I think the `filters out pendingCompaction files` is right behavior here ? If the file slice is in pending state, we would read the log files and resolve the index info there.

Reply:
but from log，pendingCompaction contains not only logs, but also parquet file(02d71f2b-bb2f-4c44-96f2-78cb791da28e_8-10-0_20220425131251.parquet), and then the logic filters the parquet,
https://github.com/apache/hudi/blob/4258a715174e0a97271112148ab20ef9307e2bd8/hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java#L422
so bootstrapOperator  cannot obtain the data in the parquet.

Reply:
and then u can run TestBoostrapOperator locally.
` .getLatestFileSlicesBeforeOrOn("par1", latestCommitTime.get().getTimestamp(), true, true)`
or
` .getLatestFileSlicesBeforeOrOn("par1", latestCommitTime.get().getTimestamp(), true, false)`

Reply:
> and then the logic filters the parquet

We should filter the parquet here, the compaction parquet and log have same content.

Reply:
However, the log file is not obtained. and

Reply:
Yes, you are right, then using `getLatestMergedFileSlicesBeforeOrOn` instead should generates correct result.